helm/superset/templates/deployment-worker.yaml

```diff
@@ -53,6 +53,9 @@ spec:
         app: {{ template "superset.name" . }}-worker
         release: {{ .Release.Name }}
     spec:
+    {{- if .Values.serviceAccount.create }}
```

Comment:
Instead of guarding this with `.Values.serviceAccount.create`, we should define the name of the SA to use in values (default of empty, or null) and then use the SA name in this condition.

Reply:
I have done exactly this in [my PR](https://github.com/apache/superset/pull/15340). Please review.